### ISSUE TITLE
FIB with FRRouting 7.2.x

### ISSUE THREAD

**Issue by dghubble** [osrg/gobgp#2228]
Upgrading from FRRouting 7.1 to 7.2, it seems connections can be made to Zebra using message version 6, but decoding failures are reported and connected addresses aren't added to the rib. I've tried setting software-name to `frr7` and `frr7.2` (which doesn't seem to be whitelisted).

```
[zebra]
  [zebra.config]
    enabled = true
    url = "unix:/var/run/frr/zserv.api"
    redistribute-route-type-list = ["connect"]
    version = "6"
    software-name = "frr7.2"
```

```
{"Topic":"Zebra","level":"info","msg":"success to connect to Zebra with message version 6.","time":"2020-01-12T20:30:40Z"}
...
{"Data":"bG8AAAAAAAAAAAAAAAAAAAAAAAAAAAABBQAAAAAAAQBJAAIAAAAAAAAAAAABAAAAAQAAAAAAAAAAAAAAAAAeAAAAAAA=","Error":"lack of bytes. need 34 but 9","Header":{"Len":78,"Marker":254,"Version":6,"VrfId":0,"Command":0},"Topic":"Zebra","level":"warning","msg":"failed to decode body","time":"2020-01-12T20:30:40Z"}
```

Anyone else tried this out? For now, workaround was to rollback to FRRouting 7.1.

### Versions

```
GoBGP: v2.12.0
FRR: 7.2 alpine
OS: custom
```

**Comment by irino** [comment]
My PR #2229 which supports FRR7.2 is merged . Could you test with gobgp master again?

**Comment by dghubble** [comment]
Thanks! With a build from #2229, I no longer see decode failures. I see the initial log about success to connect to Zebra, but no further zebra messages and connected routes aren't added to the rib. I'll try raising the log verbosity soon, but rolled back to 7.1 for now.

```
{"Topic":"Zebra","level":"info","msg":"success to connect to Zebra with message version 6.","time":"2020-01-12T20:30:40Z"}
```

If frr7.2 works for you, it could be something on my side and I'll close this issue.

**Comment by dghubble** [comment]
It seems like routes cannot be added specifically when `frr7.2` is set.

Here is a minimal configuration I used to reproduce, with all policy removed.

```
[global.config]
  as = PRIVATE_AS
  router-id = SOME_IP

  [global.apply-policy.config]
    default-import-policy = "accept-route"
    default-export-policy = "accept-route"

[zebra]
  [zebra.config]
    enabled = true
    url = "unix:/var/run/frr/zserv.api"
    redistribute-route-type-list = ["connect"]
    version = "6"
    software-name = "frr7.2"
```

Add an interface and address.

```
ip link add dev foo type dummy
ip addr add dev foo 18.18.18.18/32
ip link set foo up
```

Observe no route is added (unexpected). Explore a bit further. Try to add it manually with `gobgp`. It will be withdrawn and doesn't get added (unexpected)

```
# gobgp global rib
Network not in table
# gobgp global rib add 18.18.18.18/32
# gobgp global rib
Network not in table
```

Followup observations:

* Manually add some other IP that's not known to the kernel route table. That will work.
* Switch `software-name` back to `ffr7.1` (ignoring that the host has frr7.2 installed). Manually adding routes works again.

```
{"Topic":"Table","level":"debug","msg":"computeKnownBestPath knownPathList: 1","time":"2020-01-14T05:56:46Z"}
{"Body":{"Type":9,"Instance":0,"Flags":0,"Message":1,"SAFI":1,"Prefix":{"Family":0,"PrefixLen":32,"Prefix":"18.18.18.18"},"SrcPrefix":{"Family":0,"PrefixLen":0,"Prefix":""},"Nexthops":[{"T
ype":0,"VrfId":0,"Ifindex":0,"Gate":"0.0.0.0","BlackholeType":0,"LabelNum":0,"MplsLabels":null,"Onlink":0}],"Distance":0,"Metric":0,"Mtu":0,"Tag":0,"Rmac":[0,0,0,0,0,0],"Api":0},"Header":{"L
en":10,"Marker":254,"Version":6,"VrfId":0,"Command":8},"Topic":"Zebra","level":"debug","msg":"send command to zebra","time":"2020-01-14T05:56:46Z"}
{"Message":{"Header":{"Len":46,"Marker":254,"Version":6,"VrfId":0,"Command":33},"Body":{"Type":2,"Instance":0,"Flags":8,"Message":23,"SAFI":1,"Prefix":{"Family":2,"PrefixLen":32,"Prefix":"14
9.28.214.77"},"SrcPrefix":{"Family":0,"PrefixLen":0,"Prefix":""},"Nexthops":[{"Type":1,"VrfId":0,"Ifindex":4,"Gate":"0.0.0.0","BlackholeType":0,"LabelNum":0,"MplsLabels":null,"Onlink":0}],"D
istance":0,"Metric":0,"Mtu":0,"Tag":0,"Rmac":[0,0,0,0,0,0],"Api":33}},"Topic":"Zebra","level":"debug","msg":"read message from zebra","time":"2020-01-14T05:56:46Z"}
{"Distance":0,"Family":2,"Flag":"FLAG_SELECTED","Message":23,"Metric":0,"Mtu":0,"Nexthop":[{"Type":1,"VrfId":0,"Ifindex":4,"Gate":"0.0.0.0","BlackholeType":0,"LabelNum":0,"MplsLabels":null,"
Onlink":0}],"Prefix":"18.18.18.18","PrefixLength":32,"RouteType":"FRR_ZAPI6_ROUTE_CONNECT","Topic":"Zebra","api":"FRR_ZAPI6_REDISTRIBUTE_ROUTE_ADD","level":"debug","msg":"create path from 
ip route message.","time":"2020-01-14T05:56:46Z"}
{"Key":"18.18.18.18/32","Topic":"Table","level":"debug","msg":"Removing withdrawals","time":"2020-01-14T05:56:46Z"}
{"Body":{"Type":9,"Instance":0,"Flags":0,"Message":1,"SAFI":1,"Prefix":{"Family":0,"PrefixLen":32,"Prefix":"18.18.18.18"},"SrcPrefix":{"Family":0,"PrefixLen":0,"Prefix":""},"Nexthops":[{"T
ype":0,"VrfId":0,"Ifindex":0,"Gate":"0.0.0.0","BlackholeType":0,"LabelNum":0,"MplsLabels":null,"Onlink":0}],"Distance":0,"Metric":0,"Mtu":0,"Tag":0,"Rmac":[0,0,0,0,0,0],"Api":0},"Header":{"L
en":10,"Marker":254,"Version":6,"VrfId":0,"Command":9},"Topic":"Zebra","level":"debug","msg":"send command to zebra","time":"2020-01-14T05:56:46Z"}
{"Message":{"Header":{"Len":46,"Marker":254,"Version":6,"VrfId":0,"Command":33},"Body":{"Type":2,"Instance":0,"Flags":8,"Message":23,"SAFI":1,"Prefix":{"Family":2,"PrefixLen":32,"Prefix":"14
9.28.214.77"},"SrcPrefix":{"Family":0,"PrefixLen":0,"Prefix":""},"Nexthops":[{"Type":1,"VrfId":0,"Ifindex":4,"Gate":"0.0.0.0","BlackholeType":0,"LabelNum":0,"MplsLabels":null,"Onlink":0}],"D
istance":0,"Metric":0,"Mtu":0,"Tag":0,"Rmac":[0,0,0,0,0,0],"Api":33}},"Topic":"Zebra","level":"debug","msg":"read message from zebra","time":"2020-01-14T05:56:46Z"}
{"Distance":0,"Family":2,"Flag":"FLAG_SELECTED","Message":23,"Metric":0,"Mtu":0,"Nexthop":[{"Type":1,"VrfId":0,"Ifindex":4,"Gate":"0.0.0.0","BlackholeType":0,"LabelNum":0,"MplsLabels":null,"
Onlink":0}],"Prefix":"18.18.18.18","PrefixLength":32,"RouteType":"FRR_ZAPI6_ROUTE_CONNECT","Topic":"Zebra","api":"FRR_ZAPI6_REDISTRIBUTE_ROUTE_ADD","level":"debug","msg":"create path from 
ip route message.","time":"2020-01-14T05:56:46Z"}
{"Key":"18.18.18.18/32","Topic":"Table","level":"debug","msg":"Removing withdrawals","time":"2020-01-14T05:56:46Z"}
{"Key":"18.18.18.18/32","Topic":"Table","level":"debug","msg":"Found withdrawals for path(s) that did not get installed","time":"2020-01-14T05:56:46Z"}
```

**Comment by irino** [comment]
Thank you for reporting. I found a bug and fixed in PR #2231 .

**Comment by dghubble** [comment]
Ah! Thanks very much!
